### PR TITLE
[konflux] skip SBOM components with incorrect purl

### PR DESCRIPTION
**Pull request by ashwindasr**
purl error

```
ERROR Failed writing record to the konflux DB: Invalid purl 'pkg:github/docker:/#morphy/revive-action:v2' cannot contain a "user:pass@host:port" URL Authority component: ''.
```
Skip SBOM components with invalid purl

### CODE REVIEW

**Comment by ashwindasr** [comment]
Triggered test: https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/aos-cd-builds/job/build%252Focp4-konflux/518/console

**Comment by thegreyd** [comment]
We should ignore invalid purls, warn and then proceed with whatever we gather.

**Comment by ashwindasr** [comment]
/hold

Until we run a test

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 1]
```diff
@@ -250,16 +250,20 @@ async def _get_for_arch(arch):
...
+                    try:
+                        purl = PackageURL.from_string(x["purl"])
+                        # right now, we only care about rpms
+                        if purl.type == "rpm":
+                            # get the source rpm
+                            source_rpm = purl.qualifiers.get("upstream", None)
+                            if source_rpm:
+                                source_rpms.add(source_rpm.rstrip(".src.rpm"))
+                    except ValueError:
+                        logger.warning(f"Failed to get SBOM contents of {x['name']}")
```
Comment (thegreyd) [inline doozer/doozerlib/backend/konflux_image_builder.py:262]:
```suggestion
                    except Exception as e:
                        logger.warning(f"Failed to parse purl: {x['purl']}")
```

**Comment by ashwindasr** [comment]
/unhold

Test run succeeded

**Pull request merged**